community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/CypherException.scala

```diff
@@ -138,3 +139,19 @@ class SyntaxException(message: String, val query: String, val offset: Option[Int
 class CypherExecutionException(message: String, cause: Throwable) extends CypherException(message, cause) {
   override def mapToPublic[T <: Throwable](mapper: MapToPublicExceptions[T]): T = mapper.cypherExecutionException(message, this)
 }
+
+class ShortestPathFallbackDisabledException extends CypherExecutionException(ERROR_MSG, null) {
+  override def mapToPublic[T <: Throwable](mapper: MapToPublicExceptions[T]): T =
+    mapper.shortestPathFallbackDisableRuntimeException(message, this)
+}
+
+object ShortestPathFallbackDisabledException {
+  val ERROR_MSG: String =
+    s"""Shortest path fallback has been explicitly disabled. That means that no full path enumeration is performed in
+       |case shortest path algorithms cannot be used. This might happen in case of existential predicates on the path,
+       |e.g., when searching for the shortest path containing a node with property 'name=Emil'. The problem is that
+       |graph algorithms work only on universal predicates, e.g., when searching for the shortest where all nodes have
+       |label 'Person'. In case this is an unexpected error please either disable the runtime error in the Neo4j
+       |configuration or please improve your query by consulting the Neo4j manual.""".stripMargin
```

Comment:
What about mentioning directly here that you can separate the function and the existential predicate with a 'WITH'? It is more correct to consult the manual, but more convenient to have the answer directly in the error.